### PR TITLE
Upgrade ORC version from 1.5.8 to 1.5.10

### LINKED ISSUE #3227: [FEA] Upgrade the ORC version
Our plugin is using ORC `1.5.8` currently, however Spark 3.0 and 3.1 are using ORC `1.5.10` and `1.5.12` separately.
It will be better to upgrade it to at least` 1.5.10`.

### PR DESCRIPTION
**Pull request by res-life**
This fixes #3227

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
1. Upgrades ORC to 1.5.12
2. The "generate-build-info" execution in integration pom.xml is duplicated with root pom, so delete it.

File: integration_tests/pom.xml [hunk 1]
```diff
@@ -211,32 +211,6 @@
                     <classifier>${spark.version.classifier}</classifier>
                 </configuration>
             </plugin>
-            <plugin>
```
Comment (revans2) [inline integration_tests/pom.xml:214]: @pxLi and @tgravescs git blame indicates that the two of you touched this. It looks like it was added in on purpose so we get version info in both jars. Is that correct?

**Comment by tgravescs** [comment]
so can we have a description on this as to why we are upgrading and why did we chose 1.5.12 when Spark 3.0.1 support 1.5.10?  It probably doesn't matter anymore since we are shading it, but if it doesn't matter why aren't we upgrading to newer?

File: integration_tests/pom.xml (hunk 1 above)
Comment (tgravescs) [reply integration_tests/pom.xml:214]: Yes this is to make sure all jars have version properties which we then use later to make sure all the jars we pull are built from same version of source code, please put it back.

Comment (res-life) [reply integration_tests/pom.xml]: Reverted.

**Comment by res-life** [comment]
> so can we have a description on this as to why we are upgrading and why did we chose 1.5.12 when Spark 3.0.1 support 1.5.10? It probably doesn't matter anymore since we are shading it, but if it doesn't matter why aren't we upgrading to newer?

The default "buildver" version is 3.0.1, so let's use 1.5.10. The purpose of upgrading ORC I can see is to keep consistent when the "buildver" is 3.0.1.

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
@jlowe  @revans2  help to review.

**Review by jlowe — APPROVED** [review]

**Review by tgravescs — DISMISSED** [review]

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScanBase.scala [hunk 2]
```diff
@@ -1224,7 +1224,7 @@ private case class GpuOrcFileFilterHandler(
       val searchArg = readerOpts.getSearchArgument
       if (searchArg != null && orcReader.getRowIndexStride != 0) {
         val sa = new SargApplier(searchArg, orcReader.getRowIndexStride, evolution,
-          orcReader.getWriterVersion, useUTCTimestamp)
+          orcReader.getWriterVersion, useUTCTimestamp, false, false)
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScanBase.scala:1227]: Actually on second thought, we shouldn't be passing `false, false` here.  We should be following [what ORC 1.5.10 is doing here](https://github.com/apache/orc/blob/rel/release-1.5.10/java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java#L219-L220) which is looking up the parameters from the reader and options.

**Review by jlowe — CHANGES_REQUESTED** [review]

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScanBase.scala (hunk 2 above)
Comment (res-life) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScanBase.scala]: Updated and added test case.

**Comment by res-life** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**